File: workflow-comparisons/performance/snakemake/mock.py
import copy

def pmock(x):
    if len(x) <= 2:
        return x
    y = copy.copy(x)
    y[0], y[-1] = y[-1], y[0]
    return y

with open(snakemake.input[0], "rb") as infh:
    data = bytearray(infh.read())

modified_data = pmock(data)

with open(snakemake.output[0], "wb") as outfh:
    outfh.write(modified_data)

File: workflow-comparisons/performance/snakemake/mock.R
mock <- function(x) {
  n <- length(x)
  if (n <= 2){
      x
  } else {
      # Swap first and last characters
      temp <- x[1]
      x[1] <- x[n]
      x[n] <- temp
      x
  }
}

inputfile <- snakemake@input[[1]]
outputfile <- snakemake@output[[1]]

data <- readBin(con = inputfile, what = "raw", n = file.info(inputfile)$size)
data <- mock(data)

writeBin(data, con = outputfile)
